src/Control/Monad/Freer/Error.hs

```diff
@@ -43,12 +44,12 @@ catchError
   => Eff effs a
   -> (e -> Eff effs a)
   -> Eff effs a
-catchError m handle = interposeWith (\(Error e) _ -> handle e) m
+catchError m handle = interceptRelay pure (\(Error e) _ -> handle e) m
```

Comment:
I haven't thought too much about this, but we can probably write it without `relay`. `relay` is *relay* slow.

Reply:
Just that I didn't implement it.